common/app/common/LinkTo.scala

```diff
@@ -49,18 +50,20 @@ trait LinkTo extends Logging {
     case RssPath(path, format) =>
       ProcessedUrl(urlFor(path, edition) + format)
     case AmpPath(path, format) =>
-      ProcessedUrl(urlFor(path, edition, secure = true) + format)
+      ProcessedUrl(urlFor(path, edition, secure = true, isAmp = true) + format)
     case GuardianUrl(_, path) =>
       ProcessedUrl(urlFor(path, edition))
     case otherUrl =>
       ProcessedUrl(otherUrl, true)
   }
 
-  private def urlFor(path: String, edition: Edition, secure: Boolean = false): String = {
+  private def urlFor(path: String, edition: Edition, secure: Boolean = false, isAmp: Boolean = false): String = {
```

Comment:
It is a good thought. While working on this I have been trying to do so in a way that makes it explicit that a change is for amp. First and foremost, so it is easy to search for and remove. 

As amp continues I am weary of this approach. However, I still think it is best to wait until after launch and the business decides how they want to proceed.